internal/metrics/embeddings_metrics_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright Envoy AI Gateway Authors
```

Comment:
i think the test here can be deleted as it's duplicate of the baseMetrics method which are already tested in chat_completion_metrics_test.go.

Instead, shall we have one test for RecordTokenUsage which is the only diff from the base.

Reply:
Removed and kept only RecordTokenUsage related testcases